src/views/RegisterForm/RegisterForm.jsx

```diff
@@ -0,0 +1,342 @@
+  const [validDOB, setValidDOB] = useState(true); //no future dates
+  const [validPhone, setValidPhone] = useState(true);
+  const [validEmail, setValidEmail] = useState(true);
+  const [validPasswordFormat, setValidPasswordFormat] = useState(true);
+  const [validPassword, setValidPassword] = useState(true); //confirmed password is same
+  const [validLocation, setValidLocation] = useState(true);
+  const [userCredentials, setUserCredential] = useState({});
+  const [alertNotification, setAlertNotification] = useState({
+    isOpen: false,
+    alertType: "",
+    alertMessage: ""
+  });
+
+  const validInputs = {
+    firstName: validFirstName,
+    middleName: validMiddleName,
+    lastName: validLastName,
+    birthDate: validDOB,
+    email: validEmail,
+    phoneNumber: validPhone,
+    password: validPasswordFormat,
+    confirmPassword: validPassword,
+    location: validLocation
+  };
+
+  const nameRegex = new RegExp(/^[_A-z]*((-|\s)*[_A-z])*$/);
+  const phoneRegex = RegExp(/^[(]?[0-9]{3}[)]?[-\s]?[0-9]{3}[-\s']?[0-9]{4}$/);
+  const emailRegex = new RegExp(/[\w-.]+@([\w-]+.)+[\w-]{2,4}/);
+  const passwordRegex = new RegExp(
+    /(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[@#$%^&+*!=]).{8,20}/
+  );
+
+  /**
+   * Boolean function that returns if password matches.
+   * @param {String} password user's password
+   * @param {String} passwordToConfirm confirmation password.
+   */
+  const validatePassword = (password, passwordToConfirm) =>
+    password === passwordToConfirm;
+
+  /**
+   * This function aids the react-shard valid and invalid prop to display the errors.
+   * Won't change the color of confirm password, since confirmPassword isn't part of user credential.
+   * @param {String} name className
+   * @param {String} type valid or invalid
+   */
+  const validInvalidByName = (name, type) => {
+    if (userCredentials[[name]] && userCredentials[[name]] !== "")
+      return type === "invalid" ? !validInputs[[name]] : validInputs[[name]];
+    else return false;
+  };
+
+  /**
+   * Sets the alert or turns it off by calling the setter.
+   * @param {String} triggerType - type of trigger success or error.
+   */
+  const triggerAlert = triggerType => {
+    triggerType === "error"
+      ? setAlertNotification({
+          isOpen: !alertNotification.isOpen,
+          alertType: "danger",
+          alertMessage: "Form Not Validated"
+        })
+      : setAlertNotification({
+          isOpen: !alertNotification.isOpen,
+          alertType: "success",
+          alertMessage: "Registration Successful!"
+        });
+  };
+  /**
+   * Returns the value of the key given from the map.
+   * Map value is the result of the regex.
+   * Object that returns the proper valid input function.
+   * Lots of if statements.
+   * @param {String} name - the input name.
+   * @param {String} value - the value of the input field.
+   *
+   * */
+  const getValidationFunction = (name, value) => {
+    const validationMap = new Map();
```

Comment:
this comment this applies

https://github.com/dannylekim/billsnapFront/pull/13#discussion_r380741719